packages/nuxi/src/commands/dev.ts

```diff
@@ -90,15 +89,18 @@ export default defineNuxtCommand({
         }
 
         currentNuxt = await loadNuxt({ rootDir, dev: true, ready: false })
-        // Hard restart
-        if (process.env.NUXI_CLI_WRAPPER) {
-          currentNuxt.hooks.hook('restart', (options) => {
-            if (options?.hard) {
-              process.exit(EXIT_CODE_RESTART)
-            }
-          })
-        }
-        currentNuxt.hooks.hookOnce('restart', () => load(true))
+
+        currentNuxt.hooks.hookOnce('restart', async (options) => {
+          if (options?.hard && process.send) {
+            await listener.close().catch(() => {})
+            await currentNuxt.close().catch(() => {})
+            await watcher.close().catch(() => {})
+            await distWatcher.close().catch(() => {})
```

Comment:
Would this better to use a Promise.allSettled?

Reply:
Hmm maybe but it would make it more error prune and less predictable especially when we are unallocated os resources. I think it worth to keep it as is.